src/NumericalAlgorithms/RootFinding/GslMultiRoot.cpp

```diff
@@ -0,0 +1,151 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "GslMultiRoot.hpp"
+
+#include <exception>
+#include <gsl/gsl_multiroots.h>
+#include <gsl/gsl_vector.h>
+
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Parallel/Printf.hpp"
+
+class rootfinding_exception : public std::exception {
```

Comment:
If you want anyone to be able to catch this you need to put in in the header file.